recipes/eleuther_eval.py

```diff
@@ -163,101 +427,84 @@ class EleutherEvalRecipe(EvalRecipeInterface):
 
     Features:
         - Single GPU evaluation. Multi-GPU evaluation is currently not supported.
-        - Loading model in fp32 or bf16. Fp16 is currently not supported.
+        - Quantization and torch.compile (for text-only models) is supported.
+        - Any task from the EleutherAI eval harness
 
-    We recommend launching evaluation using the tune CLI:
+    We recommend launching evaluation using the tune CLI::
 
-        tune run eleuther_eval --config llama2_eleuther_eval \
-        tasks=["truthfulqa_mc2","hellaswag"]
-
-    Args:
-        cfg (DictConfig): OmegaConf object parsed from YAML file
+        tune run eleuther_eval --config eleuther_evaluation \
+            tasks=["truthfulqa_mc2","hellaswag"] \
+            limit=50 \
     """
 
     def __init__(self, cfg: DictConfig) -> None:
-        self._cfg = cfg
-
-    def setup(self) -> None:
-        self._device = utils.get_device(device=self._cfg.device)
-        self._dtype = training.get_dtype(dtype=self._cfg.dtype, device=self._device)
-        self._limit = self._cfg.limit
-        self._tasks = list(self._cfg.tasks)
-        self._quantizer = config.instantiate(self._cfg.quantizer)
-        self._quantization_mode = training.get_quantizer_mode(self._quantizer)
-        self._enable_kv_cache = self._cfg.get("enable_kv_cache", True)
-
-        self._batch_size = self._cfg.batch_size
-        self._max_seq_length = self._cfg.get("max_seq_length", 4096)
-
-        training.set_seed(seed=self._cfg.seed)
-
-        checkpointer = config.instantiate(self._cfg.checkpointer)
-        if self._quantization_mode is None:
+        self.device = utils.get_device(device=cfg.device)
+        self.dtype = training.get_dtype(dtype=cfg.dtype, device=self.device)
+        self.logger = utils.get_logger(cfg.get("log_level", "info"))
+        training.set_seed(seed=cfg.seed)
+
+        # Eval specific variables
+        self.limit = cfg.limit
+        self.tasks = list(cfg.tasks)
+        self.max_seq_length = cfg.max_seq_length
```

Comment:
we use "max_seq_len" everywhere else, i believe

Reply:
agree with keeping this consistent